slave/components/quality_controller/adapter_trimming.go

```diff
@@ -2,32 +2,126 @@ package quality_controller
 
 import (
 	"context"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"strings"
 
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/mount"
 	"github.com/docker/docker/client"
 
 	"github.com/genomagic/config_parser"
+	"github.com/genomagic/constants"
 )
 
 // adapterTrimming is the struct representation of the adapter trimming process
 type adapterTrimming struct {
-	// context of the process
-	ctx context.Context
 	// dockerCLI is used for launching a Docker container that perform adapter trimming
 	dockerCLI *client.Client
 	// config is the GenoMagic global configuration
 	config *config_parser.Config
+	// the context of the process
+	ctx context.Context
 }
 
-// NewAdapterTrimming constructs and returns a new instance of adapterTrimming, which implements the Controller interface
 func NewAdapterTrimming(ctx context.Context, dockerCli *client.Client, config *config_parser.Config) Controller {
 	return &adapterTrimming{
-		ctx:       ctx,
 		dockerCLI: dockerCli,
 		config:    config,
+		ctx:       ctx,
 	}
 }
 
-// Process performs the adapter trimming process by launching a Docker container that uses Trimmomatic
+// TODO: Replace this function into a constants file where this can be used by other quality control processes
+// getImageID attempts to find the Docker image by given term
+func getImageID(client *client.Client, ctx context.Context, term string) (string, error) {
+	images, err := client.ImageList(ctx, types.ImageListOptions{})
+	if err != nil {
+		return "", fmt.Errorf("failed to get available Docker images, err: %v", err)
+	} else if len(images) == 0 {
+		return "", fmt.Errorf("getImageID found no images")
+	}
+	found := false
+	assemblerID := ""
+	for _, im := range images {
+		if found {
+			break
+		}
+		for _, tag := range im.RepoTags {
+			if strings.Contains(tag, term) {
+				found = true
+				assemblerID = im.ID
+			}
+		}
+	}
+	if !found {
+		return "", fmt.Errorf("failed to find a Docker container for the given assembler")
+	}
+	return assemblerID, nil
+}
+
 func (a *adapterTrimming) Process() (string, error) {
-	return "", nil
+
+	// TrimmedFileName is the filename where the trimmed reads are going to be stored.
+	var TrimmedFileName = path.Join(constants.BaseOut, "trimmed.fastq")
+
+	img, err := getImageID(a.dockerCLI, a.ctx, "replikation/porechop")
+	if err != nil {
+		return "", fmt.Errorf("cannot get image ID for fjukstad/trimmomatic, err: %v", err)
```

Comment:
I do not know whether the error message, or the image ID is wrong here, but they are inconsistent. This is happening in the adapter trimming process, so I guess the Docker image `getImageID` should be looking for is `fjukstad/trimmomatic`.